src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -138,22 +137,32 @@ public class XpathRegressionTest extends AbstractModuleTestSupport {
             "SuppressWarningsHolder"
     );
 
-    private static Set<String> simpleCheckNames;
-    private static Map<String, String> allowedDirectoryAndChecks;
-    private static Set<String> internalModules;
+    private static final Set<String> SIMPLE_CHECK_NAMES = getSimpleCheckNames();
+    private static final Map<String, String> ALLOWED_DIRECTORY_AND_CHECKS =
+        getAllowedDirectoryAndChecks();
+
+    private static final Set<String> INTERNAL_MODULES = getInternalModules();
 
     private Path javaDir;
     private Path inputDir;
 
-    @BeforeAll
-    public static void setUpBeforeClass() throws IOException {
-        simpleCheckNames = CheckUtil.getSimpleNames(CheckUtil.getCheckstyleChecks());
+    private static Set<String> getSimpleCheckNames() {
+        try {
+            return CheckUtil.getSimpleNames(CheckUtil.getCheckstyleChecks());
+        }
+        catch (IOException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
 
-        allowedDirectoryAndChecks = simpleCheckNames
-                .stream()
-                .collect(Collectors.toMap(id -> id.toLowerCase(Locale.ENGLISH), id -> id));
+    private static Map<String, String> getAllowedDirectoryAndChecks() {
+        return SIMPLE_CHECK_NAMES
+            .stream()
+            .collect(Collectors.toMap(id -> id.toLowerCase(Locale.ENGLISH), id -> id));
```

Comment:
`id -> id` can we use identity function here?

Reply:
Done.